clean_code/maze.py

```diff
@@ -0,0 +1,74 @@
+from cell import Cell
+from position import Position
+from exceptions import BoardIntegrityError
+
+WALL = '1'
+
+
+class Maze():
+    """user-printable maze (walls and paths).
+       converts user input into a maze"""
+    def __init__(self, cells_per_row, maze_raw_form):
+        self._board = []
+        self._columns = cells_per_row
+        self._rows = int(len(maze_raw_form) / self._columns)
+        self._maze_raw_form = maze_raw_form
+
+        self._create_board_cells()
+        self._validate_maze_is_legal()
+
+    def __str__(self):
+        display = ""
+        for row in self._board:
+            for cell in row:
+                display += str(cell)
+            display += "\n"
+        return display
+
+    def get_rows(self):
+        return self._rows
+
+    def get_columns(self):
+        return self._columns
+
+    def get_board(self):
+        return self._board
+
+    def _create_board_cells(self):
```

Comment:
You can brake this function into `create_cells` and `create_board`. In the first - convert string to cells list, and in the latter  - convert the cell list to a matrix.
In this form, all function are so simple, no need to be smart to track them.